jlm/util/file.hpp

```diff
@@ -174,19 +215,42 @@ class filepath final
     return std::filesystem::is_regular_file(fileStatus);
   }
 
-  inline std::string
+  /**
+   * Creates a directory with the given filepath.
+   *
+   * @throws jlm::util::error if the filepath aleady exists, if the parent directory does not exist,
+   * or if any other filesystem error occurs
```

Comment:
You could also just say: if an error occurs. Is it necessary to explicitly mention that the file path already exist or that the parent directory does not exist.